src/lib/y2partitioner/widgets/mkfs_optiondata.rb

```diff
@@ -0,0 +1,465 @@
+            "This option is used to specify whether inode allocation is aligned. " \
+            "By default inodes are aligned as this is more efficient than unaligned access. " \
+            "But this option can be used to turn off inode alignment when the filesystem " \
+            "needs to be mountable by an old version of IRIX that does not have the " \
+            "inode alignment feature."
+          )
+        },
+
+        {
+          fs:          %i(vfat),
+          widget:      :MkfsComboBox,
+          label:       N_("FAT &Size"),
+          values:      %w(auto 12 16 32),
+          default:     "auto",
+          mkfs_option: "-F",
+          # help text, richtext format
+          help:        N_(
+            "Specify the size of the file allocation tables entries (12, 16, or 32 bit). " \
+            "If 'auto' is specified, a suitable value is chosen dependng on the file system size. " \
+            "Note that choosing an unsuitable FAT size might result " \
+            "in an error when creating the file system."
+          )
+        },
+
+        {
+          fs:          %i(vfat),
+          widget:      :MkfsComboBox,
+          label:       N_("Number of &FATs"),
+          values:      %w(auto 1 2),
+          default:     "auto",
+          mkfs_option: "-f",
+          # help text, richtext format
+          help:        N_(
+            "Specify the number of file allocation tables. " \
+            "The default is 2."
+          )
+        },
+
+        {
+          fs:          %i(vfat),
+          widget:      :MkfsInputField,
+          label:       N_("Root &Dir Entries"),
+          default:     "auto",
+          validate:    lambda do |x|
+            (x.match?(/^\d+$/) && x.to_i >= 16) || x == "auto"
+          end,
+          mkfs_option: "-r",
+          error:       N_(
+            "The minimum number of entries is 16."
+          ),
+          # help text, richtext format
+          help:        N_(
+            "Select the number of entries available in the root directory. " \
+            "Choose a value that is at least 16, or 'auto'. " \
+            "Note that the value is a lower limit and the actual root directory may be larger."
+          )
+        }
+      ]
+
+      # Remember option data.
+      #
+      # @param option [Hash]
+      #
+      def initialize(option)
+        @value = option
+        textdomain "storage"
+      end
+
+      # Read current option value from file system class.
+      #
+      # @param filesystem [Y2Storage::Filesystems]
+      #
+      # @return [String, Boolean]
+      #
+      def get(filesystem)
+        # current filesystem settings
+        fs_str = filesystem.send program + "_options"
+
+        value = default
+
+        # find option args in current settings; if it's a non-boolean
+        # option get the value from the match
+        m = fs_str.match(/(^|\s)#{Regexp.escape(option_str)}(\S*)/)
+        value = boolean? ? !default : m[2] if m
```

Comment:
Do we really need to make this line so cryptic combining `?` with an `if` at the end? I know there is a perl developer in your heart but... :wink:

Is this the same? Because I think it's, be far, more self-explanatory
```ruby
def get(filesystem)
  # current filesystem settings
  fs_str = filesystem.send(program + "_options)

  # find option args in current settings; if it's a non-boolean
  # option get the value from the match
  m = fs_str.match(/(^|\s)#{Regexp.escape(option_str)}(\S*)/)

  return default if m.nil?
  boolean? ? !default : m[2]
end
```

Reply:
The code is rubocop's suggestion, not mine, btw. So I assumed this is how rubyists like it. :-)